geomstats/information_geometry/multivariate_normal.py

```diff
@@ -9,13 +9,137 @@
 from geomstats.geometry.riemannian_metric import RiemannianMetric
+from geomstats.geometry.spd_matrices import SPDAffineMetric, SPDMatrices
 from geomstats.information_geometry.base import InformationManifoldMixin
 from geomstats.information_geometry.normal import NormalMetric
 
 
+class MultivariateNormalDistributions:
+    """Class for the multivariate normal distributions.
+
+    This class is a common interface to the following different situations:
+
+    - centered distributions
+    - distributions with diagonal covariance matrix
+    - general distributions
+
+    Parameters
+    ----------
+    n : int
+        Dimension of the sample space of the multivariate normal distribution.
+    distribution_type : str, {'centered', 'diagonal', 'general'}
+        Type of distributions.
+        Optional, default: 'general'.
+    """
+
+    def __new__(cls, n, distribution_type="general"):
+        """Instantiate class that corresponds to the distribution_type."""
+        errors.check_parameter_accepted_values(
+            distribution_type,
+            "distribution_type",
+            ["centered", "diagonal", "general"],
+        )
+        if distribution_type == "centered":
+            return MultivariateCenteredNormalDistributions(n)
+        if distribution_type == "diagonal":
+            return MultivariateDiagonalNormalDistributions(n)
+        return MultivariateGeneralNormalDistributions(n)
+
+
+class MultivariateCenteredNormalDistributions(InformationManifoldMixin, SPDMatrices):
+    """Class for the manifold of centered multivariate normal distributions.
+
+    This is the class for multivariate normal distributions with zero mean
+    on the $n$-dimensional Euclidean space. Each distribution is represented by
+    its covariance matrix, i.e. an $n$-by-$n$ symmetric positive-definite matrix.
+
+    Parameters
+    ----------
+    n : int
+        Dimension of the sample space of the multivariate normal distribution.
+    """
+
+    def __init__(self, n):
+        super().__init__(n=n)
+        self.metric = MultivariateCenteredNormalMetric(n=n)
+
+    def sample(self, point, n_samples=1):
+        """Sample from a centered multivariate normal distribution.
+
+        Parameters
+        ----------
+        point : array-like, shape=[..., n, n]
+            Symmetric positive definite matrix representing the covariance matrix
+            of a multivariate normal distribution with zero mean.
+        n_samples : int
+            Number of points to sample with each covariance matrix in point.
+            Optional, default: 1.
+
+        Returns
+        -------
+        samples : array-like, shape=[..., n_samples, n]
+            Sample from centered multivariate normal distributions.
+        """
+        geomstats.errors.check_belongs(point, self)
+        if point.ndim > 3:
+            raise NotImplementedError
+        point = gs.to_ndarray(point, to_ndim=3)
+        samples = []
+        loc = gs.zeros(self.n)
```

Comment:
NIT: maybe call this `mean` ? It seems we do not use the "location" word at all here.